Source/p_saveg.c

```diff
@@ -2684,7 +2684,11 @@ void P_UnArchiveSpecials (void)
           pusher->thinker.function = T_Pusher;
           // can't convert from index to pointer, old save version
           if (pusher->source == NULL)
+          {
           pusher->source = P_GetPushThing(pusher->affectee);
+            if (pusher->source == NULL)
```

Comment:
We need an additional check that `pusher->type == p_push`. It's perfectly fine for the other pusher types to have no source field, because they don't go through `PIT_PushThing()`, c.f. 
https://github.com/fabiangreffrath/woof/blob/4a4917c7df0445136333e5ce6e44d7fb538a7418/Source/p_spec.c#L3132